src/main/java/Duke.java

```diff
@@ -1,10 +1,68 @@
-public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+import command.*;
+import dukeexception.*;
+import parser.*;
+import storage.*;
+import task.*;
+import ui.*;
+import java.io.IOException;
+import java.util.ArrayList;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.stage.Stage;
+
+public class Duke extends Application{
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) throws IOException{
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            ui.displayLoadError();
+            tasks = new TaskList();
+        }
     }
+
+    public void run() {
+        ui.displayIntro();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readInput();
+                ui.showLine(); // show the divider line ("_______")
+                Command c = Parser.parseCommand(fullCommand);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+            } catch (DukeException e) {
+                ui.displayError(e.getMessage());
+            } catch (IOException e) {
+                ui.displaySaveError();
+            } finally {
+                ui.showLine();
+            }
+        }
+    }
+
+    @Override
+    public void start(Stage stage) {
+        Label helloWorld = new Label("Hello World!"); // Creating a new Label control
+        Scene scene = new Scene(helloWorld); // Setting the scene to be our Label
+
+        stage.setScene(scene); // Setting the stage to show our screen
+        stage.show(); // Render the stage.
+    }
+
+    
+    public static void main(String[] args) throws IOException{
+        new Duke("/Users/Simon/Desktop/Y2S2/CS2103T/duke/src/main/java/duke.txt").run();
```

Comment:
Is it possible to change the filepath to a relative path? To make it easier in case others handle this code.